chain/validator.go

```diff
@@ -16,7 +16,7 @@ func NewValidator(executor state.Applier) *Validator {
 }
 
 // ApplyMessages applies a message to a state
-func (v *Validator) ApplyMessage(context *types.ExecutionContext, state state.VMWrapper, message *types.Message) (types.MessageReceipt, error) {
+func (v *Validator) ApplyMessage(context *types.ExecutionContext, state state.VMWrapper, message *types.Message) (types.MessageReceipt, state.MinerPenaltyFIL, state.GasRewardFIL, error) {
```

Comment:
You could move ApplyResult here and return it, rather than the four values. I think that would be a little better, tho not dramatic.